src/main/java/src/chapter5/question3/Question3.java

```diff
@@ -0,0 +1,15 @@
+package src.chapter5.question3;
+
+public class Question3 {
+    public static String replaceWithX(StringBuilder string) {
+        if (string.length() < 4) {
+            return string.toString();
+        }
+        String str = "";
+        while (str.length() < string.length() - 4) {
```

Comment:
StringBuilder should be used to build the final result.

Reply:
Fixed.